### ISSUE TITLE
no SMS Messages if some of them contain From: field with special symbols

### ISSUE THREAD

**Issue by bam80** [4IceG/luci-app-sms-tool-js#9]
After installing the new version, tinkering with SIM selection and rebooting the router, can't see my massages any more:

![image](https://github.com/4IceG/luci-app-sms-tool-js/assets/233979/b5f70153-226f-4d2e-bc59-c0e60c0e2b91)

Powered by[ LuCI openwrt-23.05 branch (git-23.118.79121-6fb185f)](https://github.com/openwrt/luci) /[ OpenWrt 23.05.0-rc2 (r23228-cd17d8df2a)](https://openwrt.org/)

How could I diagnose this?

**Comment by bam80** [comment]
The messages seem are receiving because I see increasing inbox counter, just don't see the messages themselves.

**Comment by 4IceG** [comment]
There is probably some message that breaks the json structure. Show the result of the command:

sms_tool -d /dev/ttyUSB2 -r recv

Please note that the sms-tool package does not support all available types of messages.

**Comment by bam80** [comment]
Hi @4IceG, thanks for the tip.

I made a full dump of the messages, but they contain sensitive information so I can't put if here.

It was a lot of messages received since I switched the SIM, so I started to delete one by one from the end with `sms_tool`, and refresh the page with the messages.

The problematic massage was this:
```
MSG: 42
From: DITMOS
Date/Time: 08/28/23 18:06:35
Reference number: 10
SMS segment 3 of 6
участие в электронном голосовании и розыгрыше призов можно с 8 по 1
```
After I deleted it, all the messages show up in your app.

The same message in RAW format:
```
MSG: 42
07919762020041F7600DD0C42435D27C4E010008328082816053218C0500030A06030443044704300441044204380435002004320020044D043B0435043A04420440043E043D043D043E043C00200433043E043B043E0441043E04320430043D043804380020043800200440043E0437044B04330440044B044804350020043F044004380437043E04320020043C043E0436043D043E00200441002000380020043F043E00200031
```

Can we make it work for the future?

**Comment by 4IceG** [comment]
Hi @bam80 ,
Cezary is the maintainer of the sms-tool package and he will probably need the entire content of the problematic SMS sent to an e-mail.

**Comment by obsy** [comment]
But - what kind of problem? Looks like normal message.

**Comment by obsy** [comment]
@4IceG: wygląda na normalną wiadomość, bez znaków specjalnych czy innych takich.

**Comment by bam80** [comment]
Any way I could reproduce the problem, as long as I have all the dump?

**Comment by bam80** [comment]
Does the RAW contain my phone number?
If no, I could wipe only the messages with sensitive data and provide full the dump.

**Comment by obsy** [comment]
Show me:
```
/usr/bin/sms_tool -s SM -d /dev/ttyUSB2 -f  "%Y-%m-%d %H:%M " -j recv
```
Change SM and /dev/ttyUSB2 as you have. 

No, only sender number.

**Comment by bam80** [comment]
@obsy 
The message actually contains some not-plain symbols, see `^Q` in `From:` field: 
```
From: DIT^QMOS
```

Here is the hex dump:
```
00001e80  d0 b0 d1 82 d0 bd d0 be  2e 0a 0a 4d 53 47 3a 20  |...........MSG: |
00001e90  34 32 0a 46 72 6f 6d 3a  20 44 49 54 11 4d 4f 53  |42.From: DIT.MOS|
00001ea0  0a 44 61 74 65 2f 54 69  6d 65 3a 20 30 38 2f 32  |.Date/Time: 08/2|
00001eb0  38 2f 32 33 20 31 38 3a  30 36 3a 33 35 0a 52 65  |8/23 18:06:35.Re|
00001ec0  66 65 72 65 6e 63 65 20  6e 75 6d 62 65 72 3a 20  |ference number: |
00001ed0  31 30 0a 53 4d 53 20 73  65 67 6d 65 6e 74 20 33  |10.SMS segment 3|
00001ee0  20 6f 66 20 36 0a d1 83  d1 87 d0 b0 d1 81 d1 82  | of 6...........|
00001ef0  d0 b8 d0 b5 20 d0 b2 20  d1 8d d0 bb d0 b5 d0 ba  |.... .. ........|
00001f00  d1 82 d1 80 d0 be d0 bd  d0 bd d0 be d0 bc 20 d0  |.............. .|
00001f10  b3 d0 be d0 bb d0 be d1  81 d0 be d0 b2 d0 b0 d0  |................|
00001f20  bd d0 b8 d0 b8 20 d0 b8  20 d1 80 d0 be d0 b7 d1  |..... .. .......|
00001f30  8b d0 b3 d1 80 d1 8b d1  88 d0 b5 20 d0 bf d1 80  |........... ....|
00001f40  d0 b8 d0 b7 d0 be d0 b2  20 d0 bc d0 be d0 b6 d0  |........ .......|
00001f50  bd d0 be 20 d1 81 20 38  20 d0 bf d0 be 20 31 0a  |... .. 8 .... 1.|
00001f60  0a 4d 53 47 3a 20 34 33  0a 46 72 6f 6d 3a 20 44  |.MSG: 43.From: D|

**Comment by bam80** [comment]
@obsy as I said, I already deleted the badly formed message.
How could I reproduce the problem now?

**Comment by obsy** [comment]
Can you get a similar message again? Or just wait until sms are not visible again and then show the result in json.

**Comment by bam80** [comment]
It was advertisement SMS and I see this problem for the first time. Cant take a while to show up again.

**Comment by obsy** [comment]
ASCII char 11. How it should by displayed on phone? Like'
```
DIT
   MOS
```

?

**Comment by bam80** [comment]
> Show me:
> ```
> /usr/bin/sms_tool -s SM -d /dev/ttyUSB2 -f  "%Y-%m-%d %H:%M " -j recv
> ```
> Change SM and /dev/ttyUSB2 as you have.

When I use `-s SM` I see only this until I don't refresh the page with the messages again:
```
# /usr/bin/sms_tool -s SM -d /dev/ttyUSB2 -f  "%Y-%m-%d %H:%M " -j recv
{"msg":[]}
```
Without `-s` I see the output then, is the `-s` necessary and what are the variants?

**Comment by obsy** [comment]
> Without `-s` I see the output then, is the `-s` necessary and what are the variants?

-s SM
-s ME
-s TE

depending on modem and sim card.

**Comment by bam80** [comment]
> -s SM
-s ME
-s TE

@obsy 
`-s SM|ME|TE` could probably be in the help output of the app maybe?

**Comment by bam80** [comment]
> ASCII char 11, vertical tab. How it should by displayed on phone? 

I found the article about these messages, apparently they look like:
![image](https://github.com/4IceG/luci-app-sms-tool-js/assets/233979/803055cc-564a-4675-9d3a-bb3e862a48e9)
https://devicess.ru/prishla-sms-ot-dit_moslog-ili-dit_mos-eto-chto-znachit/

**Comment by obsy** [comment]
Underline.

**Comment by bam80** [comment]
Oh, no, json actually contains the same badness with ASCII 11:
```
$ ssh root@_gateway sms_tool -jd /dev/ttyUSB2 -f '"%Y-%m-%d %H:%M"' recv|hexdump -C
00003cc0  2e 22 7d 2c 7b 22 69 6e  64 65 78 22 3a 34 32 2c  |."},{"index":42,|
00003cd0  22 73 65 6e 64 65 72 22  3a 22 44 49 54 11 4d 6f  |"sender":"DIT.Mo|
00003ce0  73 4c 6f 67 22 2c 22 74  69 6d 65 73 74 61 6d 70  |sLog","timestamp|
00003cf0  22 3a 22 32 30 32 33 2d  30 38 2d 33 30 20 31 37  |":"2023-08-30 17|
00003d00  3a 35 34 22 2c 22 63 6f  6e 74 65 6e 74 22 3a 22  |:54","content":"|
00003d10  5c 75 30 34 31 38 5c 75  30 34 34 31 5c 75 30 34  |\u0418\u0441\u04|
00003d20  33 66 5c 75 30 34 33 65  5c 75 30 34 33 62 5c 75  |3f\u043e\u043b\u|
00003d30  30 34 34 63 5c 75 30 34  33 37 5c 75 30 34 34 33  |044c\u0437\u0443|
00003d40  5c 75 30 34 33 39 5c 75  30 34 34 32 5c 75 30 34  |\u0439\u0442\u04|
00003d50  33 35 20 5c 75 30 34 33  61 5c 75 30 34 33 65 5c  |35 \u043a\u043e\|
00003d60  75 30 34 33 34 20 5c 75  30 34 33 64 5c 75 30 34  |u0434 \u043d\u04|
00003d70  33 30 20 6c 6f 67 69 6e  2e 6d 6f 73 2e 72 75 2c  |30 login.mos.ru,|
00003d80  20 5c 75 30 34 33 64 5c  75 30 34 33 35 20 5c 75  | \u043d\u0435 \u|
00003d90  30 34 33 66 5c 75 30 34  33 35 5c 75 30 34 34 30  |043f\u0435\u0440|
00003da0  5c 75 30 34 33 35 5c 75  30 34 33 34 5c 75 30 34  |\u0435\u0434\u04|
00003db0  33 30 5c 75 30 34 33 32  5c 75 30 34 33 30 5c 75  |30\u0432\u0430\u|
00003dc0  30 34 33 39 5c 75 30 34  34 32 5c 75 30 34 33 35  |0439\u0442\u0435|
00003dd0  20 5c 75 30 34 33 64 5c  75 30 34 33 38 5c 75 30  | \u043d\u0438\u0|
00003de0  34 33 61 5c 75 30 34 33  65 5c 75 30 34 33 63 5c  |43a\u043e\u043c\|

**Comment by bam80** [comment]
> Does luci display it or not?

No. But see my message above.

**Comment by bam80** [comment]
So is it a problem with `sms_tool` or `luci-app-sms-tool-js`?

**Comment by obsy** [comment]
Ideologically - sms_tool (badly formatted json). But @4IceG  - can you test does luci process such json or does it return an error?

**Comment by 4IceG** [comment]
> Ideologically - sms_tool (badly formatted json). But @4IceG - can you test does luci process such json or does it return an error?

I'm trying with this message, but LuCi is not showing anything at the moment. I'll try again.

**Comment by bam80** [comment]
@4IceG please note I cut off a tail of the message with code in my dump, it's meaningless anyway

**Comment by obsy** [comment]
```
44 49 54 11 4D 4F 53 
From:DIT_MOS
Date/Time:08/28/23 20:06:35
Reference number: 10
SMS segment 3 of 6
участие в электронном голосовании и розыгрыше призов можно с 8 по 1
```

I need to read the specs to see if that's how it's supposed to work.

**Comment by obsy** [comment]
@bam80 What router are you using sms-tool on?

**Comment by bam80** [comment]
@obsy 

> Model | Kroks KNdRt31R19
Firmware Version | OpenWrt 23.05.0-rc2 r23228-cd17d8df2a / LuCI openwrt-23.05 branch git-23.118.79121-6fb185f

Modem: Quectel EP06-E

**Comment by obsy** [comment]
Mipsel. Try this version: https://dl.eko.one.pl/test/sms-tool_2022-03-21-f07699ab-1_mipsel_24kc.ipk

(uninstall old, install new, or unpack and replace binary in /usr/bin)

**Comment by obsy** [comment]
@bam80 : have you tested new sms_tool?

**Comment by bam80** [comment]
>  Try this version: https://dl.eko.one.pl/test/sms-tool_2022-03-21-f07699ab-1_mipsel_24kc.ipk

@obsy Hey hey, sorry for a long wait, I was really busy with other things.
But eventually I was in need to read my SMS and faced this problem again so I decided it's time now)

Thıs version solved a problem, could you publish the changes and make a bugfix release please?
I would like to see it in openwrt packages feed also.

Thank you very much both of you!

**Comment by bam80** [comment]
The fix itself:
https://github.com/obsy/sms_tool/commit/1b6ca03284fd65db8799dbf7c6224210093786b0

packages PR:
https://github.com/openwrt/packages/pull/22204

**Comment by bam80** [comment]
Could we please close it when the fix land in the owrt packages repo instead?
Obviously, most of the users come from pre-built owrt and don't build stuff themselves..

**Comment by helllamer** [comment]
> packages PR: [openwrt/packages#22204](https://github.com/openwrt/packages/pull/22204)

@bam80 Looks like it is not merged since 25.09.2023, MR still in WiP status. Do you need any help?

**Comment by bam80** [comment]
@helllamer it's just rebasing stuff on my side, the patch must be trivial - I would appreciate if somebody finish it.
Let me know if you need me for that, thanks.
